array_api_tests/test_statistical_functions.py

```diff
@@ -0,0 +1,365 @@
+    if type_ is bool or type_ is int:
+        msg = f"{out_repr}={out}, should be {expected} [{f_func}]"
+        assert out == expected, msg
+    elif math.isnan(expected):
+        msg = f"{out_repr}={out}, should be {expected} [{f_func}]"
+        assert math.isnan(out), msg
+    else:
+        msg = f"{out_repr}={out}, should be roughly {expected} [{f_func}]"
+        assert math.isclose(out, expected, rel_tol=0.05), msg
+
+
+@given(
+    x=xps.arrays(
+        dtype=xps.numeric_dtypes(),
+        shape=hh.shapes(min_side=1),
+        elements={"allow_nan": False},
+    ),
+    data=st.data(),
+)
+def test_min(x, data):
+    kw = data.draw(hh.kwargs(axis=axes(x.ndim), keepdims=st.booleans()), label="kw")
+
+    out = xp.min(x, **kw)
+
+    ph.assert_dtype("min", x.dtype, out.dtype)
+    _axes = normalise_axis(kw.get("axis", None), x.ndim)
+    assert_keepdimable_shape(
+        "min", out.shape, x.shape, _axes, kw.get("keepdims", False), **kw
+    )
+    scalar_type = dh.get_scalar_type(out.dtype)
+    for indices, out_idx in zip(axes_ndindex(x.shape, _axes), ah.ndindex(out.shape)):
+        min_ = scalar_type(out[out_idx])
+        elements = []
+        for idx in indices:
+            s = scalar_type(x[idx])
+            elements.append(s)
+        expected = min(elements)
+        assert_equals("min", dh.get_scalar_type(out.dtype), out_idx, min_, expected)
+
+
+@given(
+    x=xps.arrays(
+        dtype=xps.numeric_dtypes(),
+        shape=hh.shapes(min_side=1),
+        elements={"allow_nan": False},
+    ),
+    data=st.data(),
+)
+def test_max(x, data):
+    kw = data.draw(hh.kwargs(axis=axes(x.ndim), keepdims=st.booleans()), label="kw")
+
+    out = xp.max(x, **kw)
+
+    ph.assert_dtype("max", x.dtype, out.dtype)
+    _axes = normalise_axis(kw.get("axis", None), x.ndim)
+    assert_keepdimable_shape(
+        "max", out.shape, x.shape, _axes, kw.get("keepdims", False), **kw
+    )
+    scalar_type = dh.get_scalar_type(out.dtype)
+    for indices, out_idx in zip(axes_ndindex(x.shape, _axes), ah.ndindex(out.shape)):
+        max_ = scalar_type(out[out_idx])
+        elements = []
+        for idx in indices:
+            s = scalar_type(x[idx])
+            elements.append(s)
+        expected = max(elements)
+        assert_equals("max", dh.get_scalar_type(out.dtype), out_idx, max_, expected)
+
+
+@given(
+    x=xps.arrays(
+        dtype=xps.floating_dtypes(),
+        shape=hh.shapes(min_side=1),
+        elements={"allow_nan": False},
+    ),
+    data=st.data(),
+)
+def test_mean(x, data):
+    kw = data.draw(hh.kwargs(axis=axes(x.ndim), keepdims=st.booleans()), label="kw")
+
+    out = xp.mean(x, **kw)
+
+    ph.assert_dtype("mean", x.dtype, out.dtype)
+    _axes = normalise_axis(kw.get("axis", None), x.ndim)
+    assert_keepdimable_shape(
+        "mean", out.shape, x.shape, _axes, kw.get("keepdims", False), **kw
+    )
+    for indices, out_idx in zip(axes_ndindex(x.shape, _axes), ah.ndindex(out.shape)):
```

Comment:
You should be able to use the _test_stacks helper in the linalg tests for this sort of thing. Except I'm currently in the process of updating it to handle stacks along arbitrary axes, so you'll need to wait for me to finish doing that first before you can.

Reply:
This works fine as-is. I'd rather refactor after the fact myself.